File: exercism/clojure/space-age/src/space_age.clj
(ns space-age)

(def earth-orbital-period-in-seconds 31557600.0)

(def mercury-orbital-period-in-seconds (* earth-orbital-period-in-seconds 0.2408467))

(def venus-orbital-period-in-seconds (* earth-orbital-period-in-seconds 0.61519726))

(def mars-orbital-period-in-seconds (* earth-orbital-period-in-seconds 1.8808158))

(def jupiter-orbital-period-in-seconds (* earth-orbital-period-in-seconds 11.862615))

(def saturn-orbital-period-in-seconds (* earth-orbital-period-in-seconds 29.447498))

(def neptune-orbital-period-in-seconds (* earth-orbital-period-in-seconds 164.79132))

(def uranus-orbital-period-in-seconds (* earth-orbital-period-in-seconds 84.016846))

(defn on-earth [seconds]
  (/ (double seconds) earth-orbital-period-in-seconds))

(defn on-mercury [seconds]
  (/ (double seconds) mercury-orbital-period-in-seconds))

(defn on-venus [seconds]
  (/ (double seconds) venus-orbital-period-in-seconds))

(defn on-mars [seconds]
  (/ (double seconds) mars-orbital-period-in-seconds))

(defn on-jupiter [seconds]
  (/ (double seconds) jupiter-orbital-period-in-seconds))

(defn on-saturn [seconds]
  (/ (double seconds) saturn-orbital-period-in-seconds))

(defn on-neptune [seconds]
  (/ (double seconds) neptune-orbital-period-in-seconds))

(defn on-uranus [seconds]
  (/ (double seconds) uranus-orbital-period-in-seconds))

File: exercism/python/space-age/space_age.py
from enum import Enum


class SpaceAge(object):

    earth_orbital_period = 31557600
    Planet = Enum(
        'Planets', 'MERCURY  VENUS EARTH MARS JUPITER SATURN URANUS NEPTUNE')
    ratios_by_planet = {
        Planet.MERCURY: 0.2408467,
        Planet.VENUS: 0.61519726,
        Planet.EARTH: 1,
        Planet.MARS: 1.8808158,
        Planet.JUPITER: 11.862615,
        Planet.SATURN: 29.447498,
        Planet.URANUS: 84.016846,
        Planet.NEPTUNE: 164.79132}

    def __init__(self, seconds):
        self._seconds = seconds

    def do_computation(self, planet):
        return round(self._seconds / (SpaceAge.ratios_by_planet[planet] * SpaceAge.earth_orbital_period), 2)

    def on_earth(self):
        return self.do_computation(SpaceAge.Planet.EARTH)

    def on_mercury(self):
        return self.do_computation(SpaceAge.Planet.MERCURY)

    def on_venus(self):
        return self.do_computation(SpaceAge.Planet.VENUS)

    def on_mars(self):
        return self.do_computation(SpaceAge.Planet.MARS)

    def on_jupiter(self):
        return self.do_computation(SpaceAge.Planet.JUPITER)

    def on_saturn(self):
        return self.do_computation(SpaceAge.Planet.SATURN)

    def on_uranus(self):
        return self.do_computation(SpaceAge.Planet.URANUS)

    def on_neptune(self):
        return self.do_computation(SpaceAge.Planet.NEPTUNE)
